hadoop-hdds/common/src/main/java/org/apache/hadoop/util/ClosableIterator.java

```diff
@@ -17,12 +17,11 @@
  */
 package org.apache.hadoop.util;
 
+import java.io.Closeable;
 import java.util.Iterator;
 
 /**
  * An {@link Iterator} that may hold resources until it is closed.
  */
-public interface ClosableIterator<E> extends Iterator<E>, AutoCloseable {
-  @Override
-  void close();
+public interface ClosableIterator<E> extends Iterator<E>, Closeable {
```

Comment:
AutoClosable throws any exception, since we are throwing IOException better to use Closeable, moreover the previous implementation was not throwing any exceptions.

Reply:
It is better to avoid runtime exceptions as much as possible.

Reply:
1. If a client can reasonably be expected to recover from an exception, make it a checked exception. If a client cannot do anything to recover from the exception, make it an unchecked exception from [Oracle documentation]( https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html). I don't think client can do anything in this scenario.
2. `It is better to avoid runtime exceptions as much as possible.` is not true. If you look for exception in any other new language like Kotlin, they don't even have checked exception.

Reply:
done